src/checkpt.py

```diff
@@ -0,0 +1,54 @@
+"""
+Load and inspect a PyTorch Lightning checkpoint with OmegaConf compatibility.
+
+This script safely loads a PyTorch Lightning checkpoint file (.ckpt) that includes
+OmegaConf-based configuration objects (DictConfig, ListConfig). It prints out the
+top-level keys in the checkpoint, displays stored hyperparameters, and lists a
+sample of parameter names from the model's state dictionary.
+
+Usage:
+    - Set `ckpt_path` to the path of your checkpoint file.
+    - The script will print:
+        • All top-level keys in the checkpoint.
+        • The hyperparameter dictionary (if available).
+        • The number of tensors and first 20 parameter names in `state_dict`.
+
+Notes:
+    - Uses `torch.serialization.add_safe_globals()` to register OmegaConf types,
+      ensuring compatibility with Lightning's checkpoint serialization.
+    - Uses `weights_only=False` to fully load metadata, scheduler states, etc.
+    - If no `state_dict` is present, a warning is printed.
+
+Example:
+    $ python inspect_ckpt.py
+"""
+
+import torch
+import omegaconf
+import pprint
+
+ckpt_path = "logs/pretrain/runs/2025-09-26_21-15-38/checkpoints/last.ckpt"
+
+#  Allow OmegaConf globals (required for Lightning checkpoints)
+torch.serialization.add_safe_globals(
+    [
+        omegaconf.listconfig.ListConfig,
+        omegaconf.dictconfig.DictConfig,
+    ]
+)
+
+#  Force unsafe load explicitly (trusted file, your own)
+ckpt = torch.load(ckpt_path, map_location="cpu", weights_only=False)
```

Comment:
Not sure if this file is still needed since we swapped to using Hydra inits. If it is, can we replace this to instead load hydra configs and print them that way? Hydra can read in yamls and print the full structure like a json, which might achieve the same goal as this. (or, you know, we could look at the yaml directly :) ). 

If this isn't needed, suggest deleting this file.